lib/winapi/xapi.c

```diff
@@ -5,9 +5,44 @@
  */
 
 #include <xapi.h>
+#include <assert.h>
 #include <stdio.h>
+#include <string.h>
 #include <xboxkrnl/xboxkrnl.h>
 
+__attribute__((constructor)) static VOID setup_d_drive (VOID)
```

Comment:
I don't like the idea of forcing constructors on people.

One solution would be to split this into a separate library, so only if that library is linked, it would do the setup.
That way, no source-code changes are necessary (for *very* naive ports) and the user can also avoid including this code at all if they don't want it (for low-level programs for example).

Reply:
My idea was to always link this lib by default, but provide an option to *not* link it (using some Makefile variable, analogous to how we have the `NXDK_SDL=y` stuff).
Similar approaches would also still work in future build-systems probably.

We could also use weak symbols / functions in the future to have some "override" library which just stubs these kind of constructors (so the code doesn't even end up in the binary).